dependencies.py

```
@@ -1,18 +1,25 @@
+        if importlib.util.find_spec(package) is not None:
+            print(f"Package '{package}' is already installed - skipping...\n")
+            continue
+        else:
+            print(f"Package '{package}' not found - attempting to install...\n")
+
         try:
-            subprocess.run(["pip", "install", name])
-        except Exception as e:
-            raise RuntimeError(f"Unable to install {name}: {e}")
-    print(f"\nSuccessfully installed package '{name}'!\n")
+            subprocess.run(["pip3", "install", package])
```

Ruff (0.13.3)

`subprocess` call: check for execution of untrusted input

(S603)